### ISSUE TITLE
`assign_result` of  `OptimInstanceMulticrit` fails

### ISSUE THREAD

**Issue by be-marc** [mlr-org/bbotk#64]
```
library(bbotk)
source("tests/testthat/helper.R")

terminator = term("evals", n_evals = 10)
inst = MAKE_INST_2D_2D(terminator)
opt = OptimizerRandomSearch$new()
opt$optimize(inst)
```

The code throws an error because `assign_result` of  `OptimInstanceMulticrit` asserts that y is a `data.table`. The default `.assign_result` private method of `Optimizer` which is used by ` OptimizerRandomSearch` returns y as a `numeric`.

Solution would be to return y always as a `data.table` and change the assert of `assign_result` in `OptimInstance` from  `numeric` to `data.table`.

**Comment by be-marc** [comment]
Alternatively, `.assign_result` returns a numeric for singlecrit and a data.table for multicrit. I think this would be the better solution.

**Comment by be-marc** [comment]
Closed by 5a5fac77c5285ee20f1f10c97bd8bb800c9255c5
